watertap/core/tests/test_zero_order_base.py

```diff
@@ -0,0 +1,341 @@
+                index="A")
+
+        assert model.fs.unit.removal_frac_mass_solute[0, "A"].value == 0.42
+        assert model.fs.unit.removal_frac_mass_solute[0, "A"].fixed
+
+    @pytest.mark.unit
+    def test_set_param_from_data_indexed_no_entry(self, model):
+        model.fs.params.phase_list = ["Liq"]
+        model.fs.params.solvent_set = ["H2O"]
+        model.fs.params.solute_set = ["A", "B", "C"]
+        model.fs.params.component_list = ["H2O", "A", "B", "C"]
+
+        model.fs.unit = DerivedZOBase(
+            default={"property_package": model.fs.params})
+
+        model.fs.unit.removal_frac_mass_solute = Var(
+            model.fs.time, model.fs.params.solute_set)
+
+        with pytest.raises(KeyError,
+                           match="fs.unit - database provided does not "
+                           "contain an entry for removal_frac_mass_solute with "
+                           "index A for technology."):
+            model.fs.unit.set_param_from_data(
+                model.fs.unit.removal_frac_mass_solute[0, "A"],
+                {"removal_frac_mass_solute": {}},
+                index="A")
+
+    @pytest.mark.unit
+    def test_set_param_from_data_indexed_default_not_removal(self, model):
+        model.fs.params.phase_list = ["Liq"]
+        model.fs.params.solvent_set = ["H2O"]
+        model.fs.params.solute_set = ["A", "B", "C"]
+        model.fs.params.component_list = ["H2O", "A", "B", "C"]
+
+        model.fs.unit = DerivedZOBase(
+            default={"property_package": model.fs.params})
+
+        model.fs.unit.test = Var()
+
+        model.fs.unit.removal_frac_mass_solute = Var(
+            model.fs.time, model.fs.params.solute_set)
+
+        with pytest.raises(KeyError,
+                           match="fs.unit - database provided does not "
+                           "contain an entry for test with "
+                           "index A for technology."):
+            model.fs.unit.set_param_from_data(
+                model.fs.unit.test,
+                {"test": {}},
+                index="A",
+                use_default_removal=True)
+
+    @pytest.mark.unit
+    def test_set_param_from_data_indexed_use_default(self, model):
+        model.fs.params.phase_list = ["Liq"]
+        model.fs.params.solvent_set = ["H2O"]
+        model.fs.params.solute_set = ["A", "B", "C"]
+        model.fs.params.component_list = ["H2O", "A", "B", "C"]
+
+        model.fs.unit = DerivedZOBase(
+            default={"property_package": model.fs.params})
+
+        model.fs.unit.removal_frac_mass_solute = Var(
+            model.fs.time, model.fs.params.solute_set)
+
+        model.fs.unit.set_param_from_data(
+            model.fs.unit.removal_frac_mass_solute[0, "A"],
+            {"removal_frac_mass_solute": {"A": {
+                "value": 0.42, "units": "m^3/m^3"}},
+             "default_removal_frac_mass_solute": {
+                "value": 0.70, "units": "kg/kg"}},
+            index="D",
```

Comment:
I didn't catch why there is an `index="D"` here.

Reply:
Nor why it doesn't raise an error. 

If you added `assert model.fs.unit.removal_frac_mass_solute[0, "D"].value == 0.7`, I am guessing that would be True. Is the assignment of index D for "default_removal_frac_mass_solute"?

Reply:
I think the point was to use an index that didn't actually exist - there is no index D, but I am telling it to get the value for it anyway and assign that to the variable for index A (just so that I can check the result). As index does not exist, it should get the default value, rather than the specific value assigned for component A.